src/stash/src/initialize.rs

```diff
@@ -0,0 +1,626 @@
+            audit_log_event_v1::CreateClusterReplicaV1 {
+                cluster_id: DEFAULT_USER_CLUSTER_ID.to_string(),
+                cluser_name: "default".to_string(),
+                replica_name: "default".to_string(),
+                replica_id: DEFAULT_REPLICA_ID.to_string(),
+                logical_size: options.default_cluster_replica_size,
+            },
+        ),
+    ));
+
+    // Allocate an ID for each audit log event.
+    let mut audit_events_with_id = Vec::with_capacity(audit_events.len());
+    for (ty, obj, details) in audit_events {
+        let id = get_and_increment_id(tx, AUDIT_LOG_ID_ALLOC_KEY.to_string()).await?;
+        audit_events_with_id.push((id, ty, obj, details));
+    }
+
+    // Push all of our events onto the audit log.
+    const AUDIT_LOG_COLLECTION: TypedCollection<AuditLogKey, ()> =
+        TypedCollection::new("audit_log");
+    AUDIT_LOG_COLLECTION
+        .migrate_to(tx, |entries| {
+            assert!(entries.is_empty());
+
+            audit_events_with_id
+                .into_iter()
+                .map(|(id, ty, obj, details)| {
+                    let event = audit_log_key::Event::V1(AuditLogEventV1 {
+                        id,
+                        event_type: ty.into(),
+                        object_type: obj.into(),
+                        user: None,
+                        occurred_at: None,
+                        details: Some(details),
+                    });
+                    let key = AuditLogKey { event: Some(event) };
+                    MigrationAction::Insert(key, ())
+                })
+                .collect()
+        })
+        .await?;
+
+    // Set our initial version.
+    const CONFIG_COLLETION: TypedCollection<ConfigKey, ConfigValue> =
+        TypedCollection::new(USER_VERSION);
+    CONFIG_COLLETION
+        .migrate_to(tx, |entries| {
+            assert!(entries.is_empty());
+
+            vec![MigrationAction::Insert(
+                ConfigKey {
+                    key: USER_VERSION.to_string(),
+                },
+                ConfigValue {
+                    value: crate::STASH_VERSION,
+                },
+            )]
+        })
+        .await?;
+
+    Ok(())
+}
+
+pub async fn get_and_increment_id(
+    tx: &mut Transaction<'_>,
+    key: String,
+) -> Result<u64, StashError> {
+    const ID_ALLOCATOR_COLLECTION: TypedCollection<IdAllocKey, IdAllocValue> =
+        TypedCollection::new("id_alloc");
+
+    let mut id = None;
+
+    // "migrate" the ID allocator by bumping next_id for the specified key.
+    ID_ALLOCATOR_COLLECTION
+        .migrate_to(tx, |entries| {
+            let key = IdAllocKey { name: key.clone() };
+            if let Some(entry) = entries.get(&key) {
+                id = Some(entry.next_id);
+                let new_entry = IdAllocValue {
+                    next_id: entry.next_id + 1,
+                };
+
+                vec![MigrationAction::UpdateEntry(key.clone(), (key, new_entry))]
```

Comment:
I mentioned this above, but we can't use update here because the `K2` type ends up being undefined. The variant as-is wasn't super useful so I removed it